homeassistant/components/locative/__init__.py

```diff
@@ -28,13 +29,30 @@
 ATTR_DEVICE_ID = 'device'
 ATTR_TRIGGER = 'trigger'
 
-WEBHOOK_SCHEMA = vol.Schema({
-    vol.Required(ATTR_LATITUDE): cv.latitude,
-    vol.Required(ATTR_LONGITUDE): cv.longitude,
-    vol.Required(ATTR_DEVICE_ID): cv.string,
-    vol.Required(ATTR_TRIGGER): cv.string,
-    vol.Optional(ATTR_ID): cv.string,
-}, extra=vol.ALLOW_EXTRA)
+
+def _id(value: str) -> str:
+    """Coerce id by removing '-'."""
+    return value.replace('-', '')
+
+
+def _validate_test_mode(obj: Dict) -> Dict:
+    """Validate that id is provided outside of test mode."""
+    if ATTR_ID not in obj and obj[ATTR_TRIGGER] != 'test':
+        raise vol.Invalid('Location id not specified')
+    return obj
+
+
+WEBHOOK_SCHEMA = vol.All(
+    dict,
+    vol.Schema({
+        vol.Required(ATTR_LATITUDE): cv.latitude,
+        vol.Required(ATTR_LONGITUDE): cv.longitude,
+        vol.Required(ATTR_DEVICE_ID): cv.string,
+        vol.Required(ATTR_TRIGGER): cv.string,
+        vol.Optional(ATTR_ID): vol.All(cv.string, _id)
+    }, extra=vol.ALLOW_EXTRA),
```

Comment:
Do we need to allow extra?

Reply:
Not anymore, it was needed while I was getting all of the attributes in there